app/shop.py

```diff
@@ -0,0 +1,14 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list
+    products: dict
+
+    def cost_food(self, customer: callable) -> float:
+        cost_food = 0
+        for food in customer.product_cart.keys():
+            cost_food += customer.product_cart[food] * self.products[food]
+        return cost_food

```

Comment:
Use sum() function

Reply:
Сетуація така в меня два різних обьекта представникі різних классів іх обьеднує лише ключ і як зробить цю реаліцію не знаю спроба вказати ітерацію за ключами не виходить буду радий за підказку

Reply:
```suggestion
        return sum(count * self.products[food] for food, count in customer.product_cart.items())
```
You can use sum as follow)
@AnatoliyPilipey

Reply:
я зрозумів, чудове рішення

Reply:
return sum(
            customer.product_cart[food] * self.products[food]
            for food in customer.product_cart.keys()
        ) 
так також запрацювало  еще раз Дякую!!!